rankboost.py

```diff
@@ -436,6 +442,55 @@ def prepare_dataframe(mhc, predfile, expfile, pepfile, efilt_threshold, boost):
     _log.info('Finished creating the data frame for rank boosting..')
     return all_data, stats, boost
 
+def collapse_iars (stats):
+    """
+    Collapse duplicated IARs.
+    :param pandas.core.frame.DataFrame stats: The stats dataframe generated from parsing the input
+           calls
+    :return: A collapsed dataframe where overlapping IARS are merged into a single call
+    :rtype: pandas.core.frame.DataFrame
+    """
+    # Add new column for identical neoepitopes
+    stats['extended_group'] = stats.index
+    index_to_drop = set()
+    for epitope1 in stats.index:
+        # Get the IAR to compare against other IARs
+        iar_to_check = stats.loc[epitope1]['IAR_sequence']
+        for epitope2 in stats.index:
+            # If the IAR is the same or is a part of a different IAR while not being the same
+            # neoepitope, it has to be collapsed
+            if (iar_to_check in stats.loc[epitope2]['IAR_sequence']
+                and epitope1 is not epitope2):
+                # Add the neoepitope name to the extended group
+                joined_group = set()
```

Comment:
This is inefficient. For each epitope, look at lower ranks and process all matching ones outside this for loop. then drop or FLAG the ones you don't want.